content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,409 @@
+argument to `FALSE`.
+
+```{r, fig.alt = "Calibration plot with the confidence interval ribbon turned off"}
+segment_logistic %>% 
+  cal_plot_breaks(Class, .pred_good, include_ribbon = FALSE)
+```
+
+#### Rugs
+
+By default, the calibration plots include a RUGs layer at the top and at the 
+bottom of the visualization.  They are meant to give us an idea of the density
+of events, versus the density of non-events as the probabilities progress from
+0 to 1. 
+
+```{r, echo = FALSE, fig.alt = "Calibration plot with arrows pointing to where the RUGS plots are placed in the graph"}
+segment_logistic %>% 
+  cal_plot_breaks(Class, .pred_good) +
+  geom_segment(
+    x = 0.5, 
+    xend = 0.6, 
+    y = 0.9, 
+    yend = 1, 
+    arrow = arrow(length = unit(0.3, "cm")),
+    color = "#ffaaff"
+    ) +
+  geom_text(x = 0.5, y = 0.87, label = "Density of events", color = "#ff88ff")  +
+  geom_segment(
+    x = 0.5, 
+    xend = 0.6, 
+    y = 0.1, 
+    yend = 0.0, 
+    arrow = arrow(length = unit(0.3, "cm")),
+    color = "#ffaaff"
+    ) +
+  geom_text(x = 0.5, y = 0.13, label = "Density of non-events", color = "#ff88ff")
+```
+
+This can layer can be removed by setting `include_rug` to `FALSE`:
+
+```{r, fig.alt = "Calibration plot without RUGS"}
+segment_logistic %>% 
+  cal_plot_breaks(Class, .pred_good, include_rug = FALSE) 
+```
+
+## Integration with tune
+
+So far, the inputs to the functions have been data frames. In tidymodels, the tune package has methods for resampling models as well as functions for tuning hyperparameters. 
+
+The calibration plots in probably also support the results of these functions (with class `tune_results`). The functions read the metadata from the tune object, and the `truth` and `estimate` arguments automatically. 
+
+To showcase this feature, we will tune a model based on simulated data. In order
+for the calibration plot to work, the predictions need to be collected. This is
+done by setting `save_pred` to `TRUE`.
+
+```{r, eval = TRUE}
+set.seed(111)
+sim_data <- sim_classification(500)
+sim_folds <- vfold_cv(sim_data, repeats = 3)
+
+rf_mod <- rand_forest(min_n = tune()) %>% set_mode("classification")
+
+set.seed(222)
+tuned_model <- 
+  rf_mod %>% 
+  tune_grid(
+    class ~ .,
+    resamples = sim_folds,
+    grid = 4,
+    # Important: `saved_pred` has to be set to TRUE in order for 
+    # the plotting to be possible
+    control = control_resamples(save_pred = TRUE)
+  )
+
+tuned_model
+```
+
+The plotting functions will automatically collect the predictions. Each of the 
+pre-processing groups will be plotted individually in its own facet.
+
+```{r, fig.alt = "Multiple calibration plots presented in a grid"}
+tuned_model %>% 
+  cal_plot_logistic() 
+```
+
+A panel is produced for each value of `min_n`, coded with a automatically generated configuration name. This makes sure to use the out-of-sample data to make the plot (instead of just re-predicting the training set). 
```

Comment:
```suggestion
A panel is produced for each value of `min_n`, coded with an automatically generated configuration name. This makes sure to use the out-of-sample data to make the plot (instead of just re-predicting the training set). 
```